controllers/metal3.io/baremetalhost_controller_test.go

```diff
@@ -2522,3 +2552,117 @@ func TestPreprovImageAvailable(t *testing.T) {
 		})
 	}
 }
+
+// TestHostFirwmareSettings verifies that a change to the HFS
+// can be detected as it will be used to set state to Preparing
+func TestHostFirmwareSettings(t *testing.T) {
+
+	testCases := []struct {
+		Scenario   string
+		Conditions []metav1.Condition
+		Dirty      bool
+	}{
+		{
+			Scenario: "spec and status the same",
+			Conditions: []metav1.Condition{
+				{Type: "Valid", Status: "True", Reason: "Success"},
+			},
+			Dirty: false,
+		},
+		{
+			Scenario: "spec changed",
+			Conditions: []metav1.Condition{
+				{Type: "ChangeDetected", Status: "True", Reason: "Success"},
+				{Type: "Valid", Status: "True", Reason: "Success"},
+			},
+			Dirty: true,
+		},
+		{
+			Scenario: "spec invalid",
+			Conditions: []metav1.Condition{
+				{Type: "ChangeDetected", Status: "True", Reason: "Success"},
+				{Type: "Valid", Status: "False", Reason: "Success"},
+			},
+			Dirty: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Scenario, func(t *testing.T) {
+			host := newDefaultHost(t)
+			r := newTestReconciler(host)
+			i := makeReconcileInfo(host)
+			i.request = newRequest(host)
+
+			hfs := newHostFirmwareSettings(host, tc.Conditions)
+			r.Create(goctx.TODO(), hfs)
+
+			dirty, _, err := r.getHostFirmwareSettings(i)
+			if err != nil {
+				t.Fatal(err)
+			}
+			assert.Equal(t, tc.Dirty, dirty, "dirty flag did not match")
+		})
+	}
+
+}
+
+func TestTransitionToPreparing(t *testing.T) {
+	t.Run("bmh firmware change transition", func(t *testing.T) {
```

Comment:
(not blocking) Would be nice to have the usual cases setup, given that there is some duplication in the test code

Reply:
Will discuss this further but would like to do this as a followup.

Reply:
Updated tests to split TestTransitionToPreparing into 2 separate functions.